cmake/DetectOptions.cmake

```diff
@@ -68,12 +68,13 @@ function(lists_get_prefix listVars outVar)
 endfunction()
 
 # Blosc2
+option(ADIOS2_PREFER_SHARED_Blosc2 "prefer sharred Blosc2 libraries" ON)
 if(ADIOS2_USE_Blosc2 STREQUAL AUTO)
-  find_package(Blosc2 2.4)
+  find_package(Blosc2 2.10.1 CONFIG)
```

Comment:
Understand - but I think there is a preference of `find_package` to not pick up the config files if a `Find*.cmake` script is present - can we try to prefer Config files if present, e.g., if 2.10.1+ is there? Otherwise static builds won't work.
https://cmake.org/cmake/help/latest/command/find_package.html#search-modes